src/status.rs

```diff
@@ -1640,22 +452,140 @@ impl Clone for StatusCode {
     }
 }
 
-// Of the other common derivable traits, I didn’t measure them, but I guess they would be slow too.
-
 impl FromPrimitive for StatusCode {
     fn from_i64(n: i64) -> Option<StatusCode> {
-        if n < 100 || n > 599 {
+        if n < 0 || n > 65535 {
             None
         } else {
-            Some(unsafe { transmute::<u16, StatusCode>(n as u16) })
+            Some(match n {
```

Comment:
Since this is already tricky (it's possible to accidentally miss a variant in this match), it'd be best if there was only one of these lists. So here, I would defer to `from_u64`. Like: `if n < 0 { None } else { FromPrimitive::from_u64(n as u64) }`